### PR TITLE
docs(README): add `usage` section explaining all options/flags

### PR DESCRIPTION
**Pull request by cafkafk**
Signed-off-by: Christina Sørensen <christina@cafkafk.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,3 +24,45 @@ Try out: `docker run -it localhost:8080/sl:master`
...
+`--index-file-is-buildable` can be used to set if the provided index-file is a
+valid nix entrypoint by itself (i.e. don't use internal drv-wrapper).
+
+##### Authenticating over SSH
+
+To access ssh authentication based repositories, specify a ssh private key with `--ssh-private-key <sshprivatekey`.
+
+##### Substituters
+
+To use substituters for internal wharfix calls to `nix-build`, use the `--subtituters <substituters>` option to specify a comma-separated list of nix substituters.
```
Comment (srhb) [inline README.md:58]:
```suggestion
To use substituters for internal wharfix calls to `nix-build`, use the `--substituters <substituters>` option to specify a comma-separated list of nix substituters.
```

File: README.md [hunk 2]
```diff
@@ -24,3 +24,45 @@ Try out: `docker run -it localhost:8080/sl:master`
...
+root directory of the input (repo, path, derivation-output). You can change this
+by setting the `--index-file-path <indexfilepath>` option.
+
+<!-- Clarify this -->
+`--index-file-is-buildable` can be used to set if the provided index-file is a
+valid nix entrypoint by itself (i.e. don't use internal drv-wrapper).
+
+##### Authenticating over SSH
+
+To access ssh authentication based repositories, specify a ssh private key with `--ssh-private-key <sshprivatekey`.
```
Comment (srhb) [inline README.md:54]:
```suggestion
To access ssh authentication based repositories, specify a ssh private key with `--ssh-private-key <sshprivatekey>`.
```

**Review by srhb — CHANGES_REQUESTED** [review]

**Review by srhb — APPROVED** [review]

**Comment by johanot** [comment]
Thanks a lot for this :100: 

Thought: Should we break the huge cmdline api once and for all and introduce a config file instead? :) tbh, the cmdline was fine when wharfix only had a couple of different run modes, but as more and more "serve-types" and "index-types" etc. creeped in, inter-dependencies between cmdline flags became more and more confusing.

**Comment by cafkafk** [comment]
> Thought: Should we break the huge cmdline api once and for all and introduce a config file instead? :) tbh, the cmdline was fine when wharfix only had a couple of different run modes, but as more and more "serve-types" and "index-types" etc. creeped in, inter-dependencies between cmdline flags became more and more confusing.

Having them isn't inherently bad, I don't think it should be either/or, ideally wharfix would be best with both (with cmdline overriding config if both are specified). And if inter-dependencies are confusing, wouldn't they also be for a config file?

**Comment by johanot** [comment]
> And if inter-dependencies are confusing, wouldn't they also be for a config file?

From my point of view at least, some of the confusion would go away with structured attrs, like:

```nix
{
   serve = {
     type.db.connFile = "db.conn";
     indexFilePath = "foo/bar.nix";
   };
}
```

1) Not only is it easier to see what "mode" wharfix is in - from the config itself - without having the "side-effect" that wharfix actually switches modes if you specify a dbconn-file. (you could perhaps achieve the same on the cmdline with requiring explicit `--serve-type` or structuring flags into subcommands?)

2) You also get better compile time and runtime checks of the config. Using generics for sevetype and nesting required options below impl types. (you could perhaps achieve the same with moving to clap struct pattern?). The problem is currently, imho, that it's easy to forget or misspell config dependencies, e.g. `.required_unless_one(&["lulzthiswilltotallyneverfail", "repo", "dbconnfile", "derivationoutput"]))`

> ideally wharfix would be best with both (with cmdline overriding config if both are specified)

I don't know whether the extra maintenance would be worth it, honestly? I kinda have allergies for complicated config resolution. Next up: Someone comes asking for adding env vars to the mix :) In that case, I think I would prefer it simple and settle with just improving the cmdline arg structure.

Aaaaaaanyway... tldr; it was just a thought - not really important.

**Pull request merged**